src/main/java/net/fabricmc/loader/api/metadata/ModMetadata.java

```diff
@@ -16,52 +16,129 @@
 
 package net.fabricmc.loader.api.metadata;
 
-import com.google.gson.JsonElement;
-import net.fabricmc.loader.api.Version;
-
 import java.util.Collection;
 import java.util.Optional;
 
+import com.google.gson.JsonElement;
+
+import net.fabricmc.loader.api.Version;
+
+/**
+ * The metadata of a mod.
+ */
 public interface ModMetadata {
+	/**
+	 * Returns the type of the mod.
+	 *
+	 * <p>The types may be {@code fabric} or {@code builtin} by default.</p>
+	 *
+	 * @return the type of the mod
+	 */
 	String getType();
 
 	// When adding getters, follow the order as presented on the wiki.
 	// No defaults.
 
+	/**
+	 * Returns the mod's ID.
+	 *
+	 * <p>A mod's id must have only lowercase letters, digits, {@code -}, or {@code _}.</p>
+	 *
+	 * @return the mod's ID.
+	 */
 	String getId();
+
+	/**
+	 * Returns the mod's version.
+	 */
 	Version getVersion();
 
+	/**
+	 * Returns the mod's required dependencies, without which the Loader will terminate loading.
+	 */
 	Collection<ModDependency> getDepends();
+
+	/**
+	 * Returns the mod's recommended dependencies, without which the Loader will emit a warning.
+	 */
 	Collection<ModDependency> getRecommends();
+
+	/**
+	 * Returns the mod's suggested dependencies.
+	 */
 	Collection<ModDependency> getSuggests();
+
+	/**
+	 * Returns the mod's conflicts, with which the Loader will terminate loading.
+	 */
 	Collection<ModDependency> getConflicts();
+
+	/**
+	 * Returns the mod's conflicts, with which the Loader will emit a warning.
```

Comment:
I think this fails to load with theses. Not sure if theses really need to say what loader does?

Reply:
https://fabricmc.net/wiki/documentation:fabric_mod_json?s[]=conflicts

Probably should say what loader does here.